src/lib.rs

```diff
@@ -809,10 +872,20 @@ impl FormatModule {
         FormatRef(level)
     }
 
-    fn get_name(&self, level: usize) -> &str {
+    pub fn get_name(&self, level: usize) -> &str {
         &self.names[level]
     }
 
+    pub fn iter_formats(&self) -> impl Iterator<Item = (usize, Format)> + '_ {
+        (0..self.formats.len()).filter_map(|ix| {
+            let mut x_args = Vec::with_capacity(self.args[ix].len());
+            for (_, vt) in self.args[ix].iter() {
+                x_args.push(mk_value_expr(vt)?);
+            }
+            Some((ix, Format::ItemVar(ix, x_args)))
+        })
+    }
+
```

Comment:
What is this for exactly? I'm curious about the use of `mk_value_expr`.

Reply:
`mk_value_expr(vt)` is a test-helper function that maps a `ValueType` to an `Expr` with that `ValueType`. It is used to feed 1 or more dummy arguments to non-nullary ItemVars, in such a way as to not trigger complaints of incorrect Expr types before the Format itself can be type-inferred.